src/common/exif.cc

```diff
@@ -2918,7 +2918,14 @@ static void dt_set_xmp_dt_history(Exiv2::XmpData &xmpData, const int imgid, int
     snprintf(key, sizeof(key), "Xmp.darktable.history[%d]/darktable:multi_priority", num);
     xmpData[key] = multi_priority;
     snprintf(key, sizeof(key), "Xmp.darktable.history[%d]/darktable:iop_order", num);
-    xmpData[key] = iop_order;
+    // This code only writes iop_order as the default which is float, so it's wrong.
```

Comment:
Ok, let's go with that but I would remove this comment which will be not clear when the code is removed. I would say:

// we ensure that we write the iop_order as high precision...

Reply:
Fixed.